framework/include/utils/HashMap.h

```diff
@@ -32,8 +32,25 @@ class HashMap : public std::unordered_map<Key, T>
     return std::unordered_map<Key, T>::erase(k);
   }
 
-  inline bool contains(const Key & key) { return this->find(key) != this->end(); }
+  using typename std::unordered_map<Key, T>::const_iterator;
+  using typename std::unordered_map<Key, T>::iterator;
+
+  inline iterator find(const Key & k)
+  {
+    libMesh::Threads::spin_mutex::scoped_lock lock(spin_mutex);
```

Comment:
Yucko. Have we been performing concurrent read-writes to this? Or are we now?

Reply:
I don't know if we have been in the past, but I got a nasty surprise when I expected the MOOSE "`HashMap` is just a thread-safe `unordered_map`" philosophy to combine cleanly with the C++ "use `find()` instead of `operator[]` when you're not sure an entry exists".